services/media/mediaStorage.js

```diff
@@ -253,24 +282,31 @@ class MediaStorage {
                       }
 
                       let optimizing = Promise.resolve(buffer);
-                      if (data.file.mimetype === Jimp.MIME_JPEG) {
+                      if (!outputFormat && (data.file.mimetype === Jimp.MIME_JPEG)) {
                         optimizing = this.optimizeWithMozJpeg(buffer);
+                      } else if (outputFormat) {
+                        optimizing = this.convert(buffer, outputFormat);
                       }
 
                       // generate hash from resize buffer
                       // const resizeHash = await hash.hashOfFile(buffer);
 
+                      convertedMime = this.supportedConversions(outputFormat);
```

Comment:
Fix `convertedMime` not being defined.
Fix incorrect supportedConversions call, it's an object, so it can't be called as function. And you also want to assign mime to `convertedMime`.